sig-cli/contributing.md

```diff
@@ -0,0 +1,311 @@
+
+    Hello, my name is <your name>.  I would like to get involved in
+    contributing to the Kubernetes project.  What is the best way to
+    get started?
+
+Please understand that mentoring and on boarding new contributors takes
+a lot time and energy from maintainers who have many other
+responsibilities.
+
+## Before you begin
+
+- Visit the [sig-cli community page](https://github.com/kubernetes/community/tree/master/sig-cli)
+  - Join the `kubernetes-sig-cli@googlegroups.com` so you get email updates
+  - Join the Kubernetes slack channel `sig-cli`
+- Introduce yourself to the community by sending an email to kubernetes-sig-cli@googlegroups.com and let us know you want to be a contributor
+
+## Finding an existing issue to work on
+
+The preferred way to own a piece of work is to directly reach out
+to one of the sig-cli leads.  This will ensure that the issue is
+high-priority enough that it will receive PR review bandwidth from
+the sig-cli community.
+
+1. In the Kubernetes sig-cli slack channel, mention @pwittrock, @adohe, or  @fabianofranz and ask if there are any issues you could pick up
+2. Attend the sig-cli [bi-weekly meeting](https://github.com/kubernetes/community/tree/master/sig-cli).
+  - Introduce yourself at the beginning of the meeting
+
+## Life of a sig-cli bug overview
+
+    Picking up a bug and implementing a fix without discussing the scope
+    and requirements for the fix can result in wasted time
+    and effort due decisions such as the propper way to the fix the
+    underlying issue and scope of the changes.
+
+    Small 10-20 (non-test) line fixes for issues with clear solutions do not need
+    extensive discussion and can be implemented directly.
+
+    Issues requiring complex fixes or > 50 lines of (non-test) code should
+    be added to the sig-cli meeting agenda and briefly discussed
+    before an implementation PR is sent for review.  This will help
+    to identify any issues with the solution before the code review.
+
+1. Identify the bug - GitHub issue with basic description
+  - **Note:** Identify the severity and urgency of the issue.  If unknown that is ok.
+    If the issue is severe, make sure this is well communicated.
+  - Include existing workarounds in the issue
+  - Mention `@kubernetes/sig-cli-bugs` in a comment
+  - If needed, escalate to sig-cli leads on slack or at the bi-weekly meeting.
+2. Propose a solution
+  - If the issue is minor with a clear fix, skip this step and go directly to 4: Implement the code.
+  - If the issue is complex requiring non-trivial code changes, design a solution.
+    It is recommended that this [template](contributors/design-proposals/sig-cli/template.md) is used.
+  - Expect feedback from 2-3 different sig-cli community members in the form of PR comments
+  - Incorporate feedback from sig-cli community members and comment "PTAL" (please take another look)
+  - Proposal accepted by at least one sig-cli lead
+  - Proposal merged
```

Comment:
Not sure if we need proposal being merged for bugs, I think it's sufficient to discuss that within the issue itself. Only when this requires bigger thought process only then this should go through proposal process, but at that point this is rather a feature not a bug, imho.